GlobalOptions.py

```diff
@@ -2,45 +2,68 @@
 
 class GlobalOptionsBuilder(ABC):
     '''Abstract Component Class'''
+    _globalOptionsList:list=[]
+    
     @abstractmethod
     def add_options(self):
         pass
 
 class DefaultGlobalOptions(GlobalOptionsBuilder):
     '''Concrete Component Class'''
-    def __init__(self,globalOptionsList):
-        self.globalOptionsList=globalOptionsList
+    def __init__(self):
+        '''Initializing the _globalOptionsList declared in the Parent Class GlobalOptionsBuilder'''
+        self._globalOptionsList=[]
     def add_options(self):
-        return self.globalOptionsList
```

Comment:
Base class should return the list, decorators should return the composed instance